client/my-sites/plans-features-main/hooks/use-max-plan-upgrade-credits.ts

```diff
@@ -15,30 +13,37 @@ interface Props {
  * @returns {number} The maximum amount of credits possible for a given set of plans
  */
 export function useMaxPlanUpgradeCredits( { siteId, plans }: Props ): number {
-	const plansDetails = useSelector( ( state ) =>
-		plans.map( ( planName ) => ( {
-			isPlanAvailableForPurchase: isPlanAvailableForPurchase( state, siteId ?? 0, planName ),
-			planDiscountedRawPrice: getPlanDiscountedRawPrice( state, siteId ?? 0, planName ),
-			sitePlanRawPrice: getSitePlanRawPrice( state, siteId ?? 0, planName ),
-		} ) )
-	);
+	const planAvailabilityForPurchase = useCheckPlanAvailabilityForPurchase( {
+		planSlugs: plans,
+		siteId,
+	} );
+	const pricing = Plans.usePricingMetaForGridPlans( {
+		siteId,
+		planSlugs: plans,
+		storageAddOns: null,
+		coupon: undefined,
+		useCheckPlanAvailabilityForPurchase,
+		withProratedDiscounts: true,
+	} );
 
-	if ( ! siteId ) {
+	if ( ! siteId || ! pricing ) {
```

Comment:
No resolution needed for this comment:

I was curious about why the pricing object here can be falsey. For others looking at this PR, it's because plans ( and plans pricing ) data might still be loading for the `usePricingMetaForGridPlans` hook ( [link](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/plans/hooks/use-pricing-meta-for-grid-plans.ts#L119-L125) ).